parts/k8s/kubernetesconfigs.sh

```diff
@@ -420,5 +420,4 @@ configGPUDrivers() {
 ensureGPUDrivers() {
     configGPUDrivers
     systemctlEnableAndStart nvidia-modprobe || exit $ERR_GPU_DRIVERS_START_FAIL
-    retrycmd_if_failure 30 1 120 systemctl restart kubelet || exit $ERR_GPU_DRIVERS_START_FAIL
 }
```

Comment:
See https://github.com/Azure/aks-engine/pull/877/files#diff-b57d179f607a757733708864ea1bc0cbR418 to enable the nvidia runtime.

Reply:
The nvidia-docker2 package does have a file that it would normally copy to /etc/docker/daemon.json, but this would overwrite any custom settings (which I know I've seen elsewhere in the codebase)

Reply:
@cpuguy83 Is this what you're referring to?

https://github.com/Azure/aks-engine/blob/master/parts/k8s/kubernetesagentcustomdata.yml#L146

Reply:
There was somewhere that sets overlay2 as a storage driver as I recall.
But yeah that file you have referenced there is what you want.

Reply:
Extra overlay2 stuff for docker-engine has been removed.

I see what you mean above. Basically, we're currently enabling the nvidia runtime via a cloud-init-provided `/etc/docker/daemon.json`, but you're saying that the drivers installation will override that? I actually have a cluster built against this PR as-is (without your manual changes in https://github.com/Azure/aks-engine/pull/877/files#diff-b57d179f607a757733708864ea1bc0cbR418) and it looks O.K.:

```
azureuser@k8s-agent1-12179023-vmss000000:~$ cat /etc/docker/daemon.json
{
  "live-restore": true,
  "log-driver": "json-file",
  "log-opts":  {
     "max-size": "50m",
     "max-file": "5"
  }
  ,"default-runtime": "nvidia",
  "runtimes": {
     "nvidia": {
         "path": "/usr/bin/nvidia-container-runtime",
         "runtimeArgs": []
    }
  }
}
```

(Also, the E2E tests w/ cuda-vector job checks out.)